src/components/forms/DataLocation.tsx

```diff
@@ -0,0 +1,84 @@
+import React, {useCallback} from 'react';
+import {FormField, FormFieldTypes, FormValidators} from './forms';
+import {cloneDeep} from 'lodash';
+import {set} from 'lodash';
+import {Link} from 'react-router-dom';
+import {Styles} from '../../libs/theme';
+import {asIdAndDisplayText, SelectEntry} from './SelectOptionInterface';
+import {CloudProviders} from './CloudProviders';
+import {DataLocations} from "./DataLocations";
+import {ResearchStage} from "./ResearchStage";
+
+export type DataLocationInfo = {
+    cloudProvider: null | SelectEntry,
```

Comment:
Why `null` vs `undefined`? I think generally we've preferred `undefined`?

Reply:
FormField want's null, not undefined.  When undefined is passed, the input helper does not properly display.